Sources/RDeviceIdentifier/RDeviceIdentifier.swift

```diff
@@ -0,0 +1,184 @@
+            var digest = [UInt8](repeating: 0, count: Int(CC_SHA1_DIGEST_LENGTH))
+            _ = strData.withUnsafeBytes {
+                CC_SHA1($0.baseAddress, CC_LONG(strData.count), &digest)
+            }
+            deviceIdData = Data(bytes: &digest, count: Int(CC_SHA1_DIGEST_LENGTH))
+        }
+
+        guard let deviceIdData = deviceIdData,
+              let accessGroup = accessGroup,
+              let _ = try? keychain.save(data: deviceIdData, for: accessGroup) else {
+            RLogger.debug(message: "Keychain access failed")
+            return nil
+        }
+
+        _uniqueDeviceIdentifier = deviceIdData.hexString
+        return _uniqueDeviceIdentifier
+}
+
+    /// Return the model identifier of the device the application is currently running on.
+    ///
+    /// - Returns: The internal model identifier. Here is a list of some known model identifiers,
+    /// copied from the [enterprise iOS](http://www.enterpriseios.com/wiki/iOS_Devices) website.
+    /// Note that it might not be current.
+    ///
+    /// Device name                  | Model identifier(s)
+    /// ---------------------------- | -------------------
+    /// iOS Simulator                | `i386`, `x86_64`, `arm64`
+    /// Apple TV 2G                  | `AppleTV2,1`
+    /// Apple TV 3                   | `AppleTV3,1`
+    /// Apple TV 3 (2013)            | `AppleTV3,2`
+    /// iPad 2 (WiFi)                | `iPad2,1`
+    /// iPad 2 (GSM)                 | `iPad2,2`
+    /// iPad 2 (CDMA)                | `iPad2,3`
+    /// iPad 2 (Mid 2012)            | `iPad2,4`
+    /// iPad Mini (WiFi)             | `iPad2,5`
+    /// iPad Mini (GSM)              | `iPad2,6`
+    /// iPad Mini (Global)           | `iPad2,7`
+    /// iPad 3 (WiFi)                | `iPad3,1`
+    /// iPad 3 (CDMA)                | `iPad3,2`
+    /// iPad 3 (GSM)                 | `iPad3,3`
+    /// iPad 4 (WiFi)                | `iPad3,4`
+    /// iPad 4 (GSM)                 | `iPad3,5`
+    /// iPad 4 (Global)              | `iPad3,6`
+    /// iPad Air (WiFi)              | `iPad4,1`
+    /// iPad Air (Cellular)          | `iPad4,2`
+    /// iPad Air (China)             | `iPad4,3`
+    /// iPad Mini Retina (WiFi)      | `iPad4,4`
+    /// iPad Mini Retina (Cellular)  | `iPad4,5`
+    /// iPad Mini Retina (China)     | `iPad4,6`
+    /// iPhone 3GS                   | `iPhone2,1`
+    /// iPhone 4 (GSM)               | `iPhone3,1`
+    /// iPhone 4 (GSM / 2012)        | `iPhone3,2`
+    /// iPhone 4 (CDMA)              | `iPhone3,3`
+    /// iPhone 4S                    | `iPhone4,1`
+    /// iPhone 5 (GSM)               | `iPhone5,1`
+    /// iPhone 5 (Global)            | `iPhone5,2`
+    /// iPhone 5c (GSM)              | `iPhone5,3`
+    /// iPhone 5c (Global)           | `iPhone5,4`
+    /// iPhone 5s (GSM)              | `iPhone6,1`
+    /// iPhone 5s (Global)           | `iPhone6,2`
+    /// iPod touch 4                 | `iPod4,1`
```

Comment:
Lets remove that list and just link to https://www.theiphonewiki.com/wiki/Models, like:

` - Returns: The internal model identifier. See [here](https://www.theiphonewiki.com/wiki/Models) for a list of model identifiers.`